File: src/main/java/com/soapdogg/lispInterpreter/determiner/UserDefinedFunctionNameDeterminer.java
package com.soapdogg.lispInterpreter.determiner;

import com.soapdogg.lispInterpreter.datamodels.UserDefinedFunction;
import lombok.AllArgsConstructor;

import java.util.List;

@AllArgsConstructor(staticName = "newInstance")
public class UserDefinedFunctionNameDeterminer {

    public boolean determineIfUserDefinedFunctionName(
        final List<UserDefinedFunction> userDefinedFunctions,
        final String functionName
    ) {
        return userDefinedFunctions.stream().anyMatch(
            userDefinedFunction -> userDefinedFunction.getFunctionName().equals(functionName)
        );
    }
}

File: src/main/java/com/soapdogg/lispInterpreter/determiner/UserDefinedFunctionNameDeterminer.kt
package com.soapdogg.lispInterpreter.determiner

import com.soapdogg.lispInterpreter.datamodels.UserDefinedFunction

class UserDefinedFunctionNameDeterminer {
    fun determineIfUserDefinedFunctionName(
        userDefinedFunctions: List<UserDefinedFunction>,
        functionName: String
    ): Boolean {
        return userDefinedFunctions.stream().anyMatch {
             it.functionName == functionName
        }
    }
}